server/publications/spotlight.js

```diff
@@ -1,5 +1,15 @@
 Meteor.methods({
 	spotlight(text, usernames, type = {users: true, rooms: true}) {
+
+		/**
+		 * A Rocketchat callback triggered before the server search.
+		 */
+		RocketChat.callbacks.run('onSpotlightStart', {
```

Comment:
@lobo87 I expected {text, accessibleRoomIds}. All the rest is not subject to the external search provider

Reply:
well this "accessibleRooms" is something which is smarti specific, therefore it is being calculated in the hook implemntation. This way, the core hook can stay as general as possible, and people can start using it, in a way they want to.